src/export/InstanceExporter.ts

```diff
@@ -226,6 +226,29 @@ export class InstanceExporter implements Fishable {
             break;
           }
         }
+        // if we don't have instanceChild yet, it may be due to a rule that referes to a named slice somewhere before the end of the path using a numeric index.
+        // check for a possible choice slice that doesn't include the slice name that will satisfy the cardinality.
+        // warn the user that it is preferable to use the name of a slice whenever possible.
+        // eventually, this logic can be removed when array indexing no longer allows the author to refer to
+        // a named slice using a numeric index.
+        if (instanceChild == null) {
+          const namelessChoiceSlices =
+            element
+              .findConnectedSliceElement()
+              ?.children(true)
+              .filter(c => c.path === child.path && c.sliceName) ?? [];
+          for (const choiceSlice of namelessChoiceSlices) {
+            instanceChild = instance[choiceSlice.sliceName];
```

Comment:
This is kind of some complex stuff to wrap one's head around.  I noticed that this code is specialized to work w/ choices (which makes sense since it is in a block that checks for a path ending with `[x]`).

But here is my question: does this issue only happen for choice slices?  The example in the filed bug happens to be on a path w/ a choice slice, but I don't know if that is coincidence or if the choice slice is actually part of the recipe for reproducing.

Reply:
I did actually mess with the example some and try to reproduce the issue w/ a non-choice element -- and I _couldn't_.  So I guess it is just specific to choices.  But I still don't 100% understand why.

Our instance implementation is super tricky.